torchvision/models/detection/fcos.py

```diff
@@ -636,19 +643,37 @@ def forward(
         return self.eager_outputs(losses, detections)
 
 
-model_urls = {
-    "fcos_resnet50_fpn_coco": "https://download.pytorch.org/models/fcos_resnet50_fpn_coco-99b0c9b7.pth",
-}
+class FCOS_ResNet50_FPN_Weights(WeightsEnum):
+    COCO_V1 = Weights(
+        url="https://download.pytorch.org/models/fcos_resnet50_fpn_coco-99b0c9b7.pth",
+        transforms=ObjectDetection,
+        meta={
+            "task": "image_object_detection",
+            "architecture": "FCOS",
+            "publication_year": 2019,
+            "num_params": 32269600,
+            "categories": _COCO_CATEGORIES,
+            "interpolation": InterpolationMode.BILINEAR,
```

Comment:
Hi @datumbox ,

I just have one question here.

Seems that `GeneralizedRCNNTranform` also plays some roles of the pre-processing (pre-transforms) for an object detection task, could we set aside some attributes into the `WeightsEnum` class for further modifications or adjustments?

For example, we may also have different implementations in `GeneralizedRCNNTranform`, depending on the scaling and padding modes. And the current implementation of `GeneralizedRCNNTranform` is limited to only one padding mode (we put the image on the left-top plate). So could we make this Class more powerful and more general ?

Reply:
Thanks, great point. 

Ideally, we should explore if we can make the Object detection models to work similar to Classification, Segmentation, Optical Flow etc by removing the specific transforms from within the model. If you see PR #5444 I'm exploring adding the ability to turn `GeneralizedRCNNTranform.resize()` off to allow for more advanced augmentation strategies such as Large Scale Jitter etc. There are pros/cons for that and this is a broader investigation that I would actually like to get your input.

Concerning adding more fields to the meta, I think we should not **yet** add more until we actually decide what to do. Once this is clear to us, we will just add them. You can debate that interpolation is in the meta while not directly configurable, but the only reason it's there is for consistency across other models. Not sure if that makes sense to you. Please let me know your thoughts.